### ISSUE TITLE
Trailing Parameter

### ISSUE THREAD

**Issue by generikvault** [dart-lang/language#4079]
Allow Optional Syntax `T(a:) v;` instead of `T(a:v);` for the the final named parameter in Dart Function Calls.

Deeply nested code, such as Flutter widget trees, can lead to excessive indentation and numerous closing parentheses at the end of expressions. This proposal introduces an optional syntax to address these issues.

**Example: Current Syntax**

``` dart
Widget build(BuildContext context) {
  return Container(
    padding: EdgeInsets.all(8.0),
    child: Column(
      children: [
        Text('Hello'),
        Text('World'),
      ],
    ),
  );
}
```

**Proposed Syntax with Trailing Named Parameter**

``` dart
Widget build(BuildContext context) {
  return Container(
    padding: EdgeInsets.all(8.0),
  child:) Column( // indentation of this line reduced because it closes Container
  children:) [ // indentation of this line reduced because it closes Column
    Text('Hello'),
    Text('World'),
  ];
}
```

In this example, the code structure is flatter, reducing both indentation and the visual clutter of closing parentheses.

## Backwardompatibility

This proposal is fully backward-compatible. The existing syntax remains unchanged and valid. The new syntax is purely additive and optional, giving developers the flexibility to use it as needed.

**Comment by lrhn** [comment]
The number of down votes pregnant reflect that the suggested syntax isn't that readable. 
It avoids some indentation (not entirely clear how or why it's formatted like that), but breaks the grouping, making a `)` no longer mark the end of the expression.

**Comment by hydro63** [comment]
The biggest problem with making this happen is the complete unreadibility of the hierarchy. It's not clearly visible which property is in what widget. That is really bad to do, since two completely different structures look almost the same, making it more annoying to work with it.

**Comment by Wdestroier** [comment]
The proposed syntax fails at solving the problem of removing the numerous closing parenthesis and brackets. It looks malformatted too, if you remove all indentation whitespace, then it looks more readable.

I made a blend between YAML and Dart:
![image](https://github.com/user-attachments/assets/612fea7b-c7e8-46a8-b57f-da9fe8875c34)

Constantly reformatting this would probably be tiring. The syntax is almost certainly ambiguous with other Dart features too.

Could be easier to create a feature to "Collapse closing parenthesis".
![image](https://github.com/user-attachments/assets/03425635-009e-4159-ba5e-58a977ccae39)

Another alternative could be to show more info about the line which opens the closing parenthesis. However, it's not great to read the code in reverse order, it's mostly useful for long code and the visual pollution increases by a factor.
![image](https://github.com/user-attachments/assets/9df7b2b3-c9d1-4c82-af14-1b648759652c)

**Comment by generikvault** [comment]
> The number of downvotes probably reflect that the suggested syntax isn't that readable. It avoids some indentation (not entirely clear how or why it's formatted like that), but breaks the grouping, making a `)` no longer mark the end of the expression.

Thanks for the feeback. I should have explained my thoughts behind the indentation. I reduced the indentation on lines with a trailing prameter. 

So all but the last parameter are grouped by their indentation.

**Comment by rrousselGit** [comment]
Note that the formatter is being updated to reduce indentation.   
It removes trailing commas in various places, to remove a level of indentation.

**Comment by hydro63** [comment]
> Could be easier to create a feature to "Collapse closing parenthesis".

That could definately be possible, and not even all that complicated implement. If we are talking purely about the estetics, the LSP could just mark it as collapsable, and it would only change how the file is showed (sort of like collapsing the functions, and conditions);

_This is just a thought experiment_
If we are talking about a change in compiling process, the lexer could keep stack of all the `[(`, and when it encounters `...`, it could just expand the needed tokens to complete them.

But that leaves around the question of where should the compiler stop expanding. You can't really specify how many levels the `...` is collapsing, making a lot of ambiguous edge-cases. Firstly, what happens if there are properties after that? What happens if the object is enclosed in a list?

Basically, making it a feature of the LSP, rather than the feature of the compiler and language is probably the way to go.
